packages/core/src/browser/widget-open-handler.ts

```diff
@@ -93,6 +93,14 @@ export abstract class WidgetOpenHandler<W extends BaseWidget> implements OpenHan
             await this.shell.activateWidget(widget.id);
         } else if (op.mode === 'reveal') {
             await this.shell.revealWidget(widget.id);
+            this.shell.currentChanged.emit({
```

Comment:
only shell can fire such events via the widget tracker

Reply:
Also it is bogus, revealing the widget does not change the focus and should not change the current widget

Reply:
Thank you for the review, do you have any suggestions as to how will we be able to fix this issue?

Reply:
Sorry, I still don't understand the issue: https://github.com/eclipse-theia/theia/issues/7170#issuecomment-591298541. As I've mentioned already the editor preview should not be focused. It is correct.

cc @AlexTugarev @vince-fugnitto

Reply:
@akosyakov try to preview a file in VS Code, even though the focus is not set to the opened preview, but the status bar will correctly show the current language and information for said editor.

Right now in Theia the status bar uses `shell.currentWidget` to get its information from, and it seems to be driven by the `FocusTracker` instance, except we are not sure if we should register preview editors with it?

Reply:
Understood, I don't think we can change the semantic of `shell.currentWidget` but I wonder whether we can build on top if a different semantic for a current widget of some kind: https://github.com/eclipse-theia/theia/issues/7170#issuecomment-597546369